nomos-utils/src/overwatch_recovery.rs

```diff
@@ -0,0 +1,286 @@
+// STD
+use std::fmt::Debug;
+use std::io;
+use std::marker::PhantomData;
+// Crates
+use log::error;
+use overwatch_rs::services::state::{ServiceState, StateOperator};
+use serde::de::DeserializeOwned;
+use serde::{Deserialize, Serialize};
+
+trait FromSettings {
+    type Settings;
+    fn from_settings(settings: &Self::Settings) -> Self;
+}
+
+#[derive(thiserror::Error, Debug)]
+pub enum MyError {
+    #[error(transparent)]
+    IoError(#[from] io::Error),
+    #[error(transparent)]
+    SerdeError(#[from] serde_json::Error),
+}
+
+type _Result<T> = Result<T, MyError>;
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+struct RecoverableState<Backend, Settings>
+where
+    Backend: RecoveryBackend + Debug + Clone,
+    Backend::State: Serialize + DeserializeOwned + Debug + Clone,
+{
+    #[serde(flatten)]
+    state: <Backend as RecoveryBackend>::State,
+    #[serde(skip)]
+    settings: PhantomData<Settings>,
+    #[serde(skip)]
+    backend: PhantomData<Backend>,
+}
+
+impl<Backend, Settings> RecoverableState<Backend, Settings>
+where
+    Backend: RecoveryBackend + Debug + Clone,
+    Backend::State: Serialize + DeserializeOwned + Debug + Clone,
+{
+    fn new(state: <Backend as RecoveryBackend>::State) -> Self {
+        Self {
+            state,
+            settings: Default::default(),
+            backend: Default::default(),
+        }
+    }
+}
```

Comment:
Is this even needed? Also how does settings and backend are not bind to Default, but this still works 🤔 ?

Reply:
The compiler didn't complain about the `Default` trait for them.
Well, I was pretty sure I was using them somewhere but maybe I removed the usage.